src/controller/CHIPDeviceController.cpp

```diff
@@ -3163,6 +3189,56 @@ void DeviceCommissioner::PerformCommissioningStep(DeviceProxy * proxy, Commissio
         }
     }
     break;
+    case CommissioningStage::kGetTCAcknowledgments: {
+        ChipLogProgress(Controller, "Get Terms and Conditions Acknowledgments");
+
+        bool acknowledgementReady       = params.GetTermsAndConditionsAcknowledgement().HasValue();
+        bool acknowledgementNotRequired = params.GetRequireTermsAndConditionsAcknowledgement().HasValue() &&
```

Comment:
where is the logic for determining whether the acknowledgement is required based upon reading the current attribute values?

Reply:
The logic is in the other PR: https://github.com/project-chip/connectedhomeip/pull/34065